lib/manager/composer/artifacts.ts

```diff
@@ -97,28 +97,28 @@ export async function updateArtifacts(
     }
     const env = getChildProcessEnv(['COMPOSER_CACHE_DIR']);
     const startTime = process.hrtime();
-    let cmd: string;
+    const cmd = 'composer';
+    const execOptions: ExecOptions = {
+      cwd,
+      env,
+    };
     if (config.binarySource === 'docker') {
       logger.info('Running composer via docker');
-      cmd = `docker run --rm `;
-      if (config.dockerUser) {
-        cmd += `--user=${config.dockerUser} `;
-      }
-      const volumes = [config.localDir, process.env.COMPOSER_CACHE_DIR];
-      cmd += volumes.map(v => `-v "${v}":"${v}" `).join('');
-      const envVars = ['COMPOSER_CACHE_DIR'];
-      cmd += envVars.map(e => `-e ${e} `);
-      cmd += `-w "${cwd}" `;
-      cmd += `renovate/composer composer`;
+      execOptions.docker = {
+        image: 'renovate/composer',
+        tag: null,
+        dockerUser: config.dockerUser,
+        volumes: [config.localDir, process.env.COMPOSER_CACHE_DIR],
+        envVars: ['COMPOSER_CACHE_DIR'],
```

Comment:
Are we at risk of mismatches between `execOptions.env` and `execOptions.docker.envVars`? I guess there are times we might want to pass an env variable to child_process but not to `docker run`, but I can't think of any. Can you check if we have any use cases yet? If not then we could combine the two.

Reply:
I'll write more mocks handling env-vars machinery and the combine these parameters